### DISCUSSION TITLE
Download from login-protected page (get cookies from Firefox)

### DISCUSSION THREAD

**Discussion by jocap** [setvisible/DownZemAll#215] [Ideas]
One of the benefits of having the download manager integrated into Firefox is that it can access pages that are protected by a cookie-based login. This is the only case I've found where DownZemAll, as a standalone application, is limited.

Ideally, the Firefox add-on would relay relevant cookies to DownZemAll.

Judging from the [documentation], it seems rather simple to access cookies from a web extension. This code should (I think) access the current tab's cookies, provided the extension has the correct permissions:

    browser.cookies.getAll({url: tabs.pop().url}).then((cookies) => {
        for (let cookie of cookies) {
            // cookie.name
            // cookie.value
        }
    })

[documentation]: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies

Then it's a matter of sending that to DownZemAll and having DownZemAll make the request using the cookies.

**Comment by setvisible** [comment]
Hi,
Looks like a challenging feature to implement!

**Comment by setvisible** [comment]
Moved to "Discussions" to see how many people are interested
